File: Codewars/Reverse polish notation calculator.cs
// Reverse polish notation calculator
// https://www.codewars.com/kata/reverse-polish-notation-calculator/
// by Equim on 16-12-16

using System;
using System.Collections.Generic;
using System.Linq;

public class Calc
{
    public double evaluate(string input)
    {
        if (input == "") return 0D;

        var Expr = input.Split(' ');
        Stack<double> num = new Stack<double>();

        foreach (var item in Expr)
        {
            switch (item)
            {
                case "+": num.Push(num.Pop() + num.Pop()); break;
                case "-": num.Push(-num.Pop() + num.Pop()); break;
                case "*": num.Push(num.Pop() * num.Pop()); break;
                case "/": num.Push(1D / num.Pop() * num.Pop()); break;
                default: num.Push(double.Parse(item)); break;
            }
        }
        return num.Pop();
    }
}

File: Codewars/Reverse polish notation calculator.js
// Reverse polish notation calculator
// https://www.codewars.com/kata/reverse-polish-notation-calculator/
// by Equim on 16-12-16

function calc(input)
{
    if (input == "") return 0;

    var Expr = input.split(' ');
    var num = [];

    for (var item in Expr)
    {
        switch (Expr[item])
        {
            case "+": num.push(parseFloat(num.pop() + num.pop())); break;
            case "-": num.push(parseFloat(-num.pop() + num.pop())); break;
            case "*": num.push(parseFloat(num.pop() * num.pop())); break;
            case "/": num.push(parseFloat(1 / num.pop() * num.pop())); break;
            default: num.push(parseFloat(Expr[item])); break;
        }
    }
    return num.pop();
}